torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1444,6 +1444,284 @@ static std::vector<const Var*> getOuterLoopIndexes(Stmt* s) {
   return res;
 }
 
+class CacheReplacer : public IRMutator {
+ public:
+  CacheReplacer(
+      const Buf* buffer,
+      const Buf* cache,
+      std::vector<const Expr*>& offsets)
+      : buf_(buffer), cache_(cache), offsets_(offsets) {}
+
+ private:
+  const Expr* mutate(const FunctionCall* v) override {
+    Function* func = v->tensor()->function();
+    const Buf* buf = v->tensor()->buf();
+    if (buf != buf_) {
+      return IRMutator::mutate(v);
+    }
+
+    // for reductions the size of func->args() is not equal to the size of the
+    // output buffer, but they should be ordered so that the output args are at
+    // the beginning even if the loops are reordered later.
```

Comment:
If they are reordered, should we bail from this transformation?

Reply:
I don't think we can tell if they are or not, but I believe I've verified that they cannot be.

@ZolotukhinM I've just realized we rely on the Function() object for this - does this still work after your PR removing it?

Reply:
All the data was moved from `Function` to `Tensor` in that PR so while some details could change, what was possible before would remain possible after.